src/static_array.cr

```diff
@@ -67,31 +111,74 @@ struct StaticArray(T, N)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block at the given index.
```

Comment:
Should be something like: "Yields the current element at the given *index* and updates the value at the given *index* with the block's value"